src/root/async.c

```diff
@@ -99,6 +99,11 @@ int AsyncRead::read(size_t i)
 
 void AsyncRead::dispose(AsyncRead *aw)
 {
+    for (size_t i = 0; i < aw->filesdim; i++)
+    {
+        FileData *f = &aw->files[i];
+        WaitForSingleObject(f->event, INFINITE);
+    }
     free(aw);
```

Comment:
Can we remove this part of the change? It's asymmetric to the POSIX version and there is no need to call dispose while still reading. You could make it an assertion that read was called for each file.